### ISSUE TITLE
Does not compile. 

### ISSUE THREAD

**Issue by OrbitAVI** [MarlinFirmware/Marlin#4413]
OK please excuse my ignorance. I have been maintaining my own fork of Marlin Bug Fix for a while now, so I think I can navigate around it ok, but am no expert by any means and am pretty new to 3D printing as a whole. 

I cannot get this to compile with my printer params (basic stuff like PID settings, steps, etc..). I get an error. If I change nothing I also get a compile error. Printer is a i3.  Any help would be greatly appreciated and again sorry for being a noob at this.

   stepper.synchronize();
   ^
exit status 1
'Planner' does not name a type

**Comment by OrbitAVI** [comment]
OK I'm dumb. With the example config (ROXY) I can compile, but with errors. I figured the base config.h file was OK to use.

**Comment by Roxy-3D** [comment]
Can you post your Configuration.h file please?    I'll get that resolved ASAP.    But in the mean time, please use my Configuration.h file and cross your settings over to it.   It may be you have some #define's that conflict with each other.    

Update:  By My Configuration.h file, I mean look in the Examples folder for a folder with Roxy in it.   Those are my Configuration files and if you limit the number of changes you have to make to just what you can live with...  We have the best chance of getting you running quickly.

**Comment by OrbitAVI** [comment]
I was able to get it t compile using your example config and adding my settings however I did still get errors or warnings. I also got an error about only defining one type of probe. I only uncommented the fixed probe, but some of the servo options were conflicting, so I had to comment those out too. I'll post my config file later. Thanks for your help.

**Comment by OrbitAVI** [comment]
I was able to compile now with only warnings. The thing that threw me off using your example config is that you moved the servo probe out of the Z probe section. I'm using a fixed mounted probe. Also you had TEMP_SENSOR_0 defined twice.

This config file is compiling for me with warnings (sketch\Marlin.h:51:37: warning: extra tokens at end of #ifdef directive [enabled by default]). I did update config_adv, however the only option I use there is dual Z steppers.

 I'll load it up and try it out tonight. Can't wait! Thank you

``` cpp
/**
 * Marlin 3D Printer Firmware
 * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
 *
 * Based on Sprinter and grbl.
 * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
 *
 * This program is free software: you can redistribute it and/or modify

 * Configuration.h
```

**Comment by Roxy-3D** [comment]
@OrbitAVI  Thank's for the response.    Currently you need to have both a LCD Panel and Z-Probe defined.   I've added the corresponding checks for that in the Sanity Checks file.   Long term, both of those constraints can go away.   But we have some re-organization of the code to do before we can relax those requirements.

**Comment by OrbitAVI** [comment]
I have both defined in mine with a fixed probe and REPRAP_DISCOUNT_SMART_CONTROLLER or am I missing something? Again thank you very much.

**Comment by Roxy-3D** [comment]
@OrbitAVI I cut and pasted your posted Configuration.h file into the build environment.  The compiler complained about some unrecognizable characters (picked up from the cut and paste) but everything compiled clean with the current source.    Can you try your settings against the current code?   I've added a few changes to the Sanity Checks and I've pulled in more #include files in a few of the .cpp files.   That might get you compiling now.

**Comment by OrbitAVI** [comment]
I can compile with warnings using the latest code, however if I enable SD Support I cannot compile.

sketch\ultralcd.cpp: In function 'void lcd_sdcard_stop()':

ultralcd.cpp:567: error: 'cancel_heatup' was not declared in this scope

```
   cancel_heatup = true;

   ^
```

exit status 1
'cancel_heatup' was not declared in this scope

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.
`

**Comment by Roxy-3D** [comment]
@OrbitAVI Can you try the very latest?   I think it is currently a compiler dependency we are fighting.  I made a change that might fix it.

**Comment by OrbitAVI** [comment]
Same error if SD support is enabled. Otherwise no issues. Thank you again for taking the time to support. Personally I can get by without SD support for now and just print straight from Octoprint.
'
exit status 1
'cancel_heatup' was not declared in this scope
'

**Comment by Roxy-3D** [comment]
@OrbitAVI 
OK...   Let me take a quick look at that.    Also, it might be helpful for you to be using Arduino v1.6.8 instead of v1.6.9     But I've got v1.6.9 loaded now so I can more easily duplicate your problems.   Let me see what I can find out about SD Memory Card support.

I'll report back soon.

**Comment by OrbitAVI** [comment]
Same error in v1.6.8.

Thank you

**Comment by Roxy-3D** [comment]
@OrbitAVI 
I think I might have it fixed in both v1.6.8 and v1.6.9    

I can compile clean with SD Support turned on now.

**Comment by OrbitAVI** [comment]
Compiled with the latest + SD Support. The latest code gives the warnings again, but everything is all good. Running phase 1 now.

**Comment by Roxy-3D** [comment]
:)    When you get a G26 to run successfully....   Please post a picture for people to see!    And if you have any questions, or things that don't quite make sense to you....  Please don't hesitate to just ask!   You have already struggled enough.  Let's make the rest of the process easy and fun for you!!!!

(And don't spend a lot of time trying to get the unreachable positions 'correct'.   Just get some ball park numbers in for them.   Once you get a G26 printed, G29 P4 R will help you get all of those points adjusted very accurately and very quickly.)

**Comment by OrbitAVI** [comment]
Thanks for your help. I do have some questions. Should I close this and start a new thread?

**Comment by Roxy-3D** [comment]
Either way...   But it might be best to start a new issue with a title that matches your question.   Hopefully, you can get a G26 to print a test pattern.   If so, you will be able to calibrate your Mesh very quickly now.

**Comment by OrbitAVI** [comment]
Thanks again for the awesome support! Closing this issue as I am able to compile and flash.
